### PR TITLE
Ci/integration tests

### PR DESCRIPTION
**Pull request by 2e0byo**
This PR migrates to poetry and adds integration tests.  It actually doesn't test for deprecation warnings (yet) as I wasn't able to trigger any.  The warning I noticed can't be caught I think as we override the method, so I've just fixed it as part of the PR.

These tests are 'user stories' in the web world. Currently there are two:

- user loads up default config.  Mopidy pauses and generates a tidal login url.
- user loads up lazy config.  Mopidy does not generate a tidal login url.  User accesses tidal (via `mpc list artist`).  Mopidy generates a tidal url.

FWIW I think you'll find the user-story style tests with pexpect easier to read.  I've never used it before and was pleased how easy it was.

We could have other user-story tests if we had a mock tidalapi, but tbh I think we've got enough users.  This just stops people (=me) PRing hopelessly broken builds.

I've deliberately built this at the highest level, i.e. actual subshells.  It would be possible to call into mopidy ourselves, perhaps by instantising `core`, and get better access to the internals.  But I don't want to learn that much about Mopidy tonight.

We test against all major-numbered supported Mopidy versions + latest upstream git and all major-number supported python versions.  With the git test we should get some kind of feedback against upstream.  Thus I've set this to run every week.

# Poetry

I feel I owe an explanation of poetry as I'm suggesting my preferred tooling.  I've updated the readme, but to setup the venv you just do `poetry shell`.  If you want to be able to run mopidy from inside the venv, you can either use system-site-packages like before or just install pygobject (which I prefer, as venvs are supposed to decouple stuff, and we break that as soon as we use system-site-packages). Unfortunately poetry's support for system-site-packages is currently broken... there's a hack in the readme/makefile but <sup><sub>it involves making the venv the old way</sub></sup>.  Ahem.

I *still* think we should move to poetry:

- one config file!
- all the cool kids use it!
- easy releases to pypi!  (just run `poetry release` and it will do the rest)
- really easy dependency management
- reproducible builds + dev environment (the devtools are installed inside the venv, so we have the same versions)

Still I don't want to break anyone's workflow.

# PS

I few other things have sneaked through.  I'm currently renaming tests with the sentence-style names, can't remember if I've mentioned that  before.  Personally I think it adds a lot of clarity.  I mean to base my next PRs off this whenever they land, so if you are happy with everything just merge the most recent.  But we can keep discussion of this here.

### CODE REVIEW

**Comment by 2e0byo** [comment]
Closes #111.  Also slightly improved the ci and added caching, which saves a whole *two seconds* on average on the test run.

File: .github/workflows/test.yml [hunk 1]
```diff
@@ -11,31 +11,31 @@ jobs:
...
+        uses: actions/setup-python@v4
         with:
           python-version: ${{ matrix.python-version }}
+          cache: 'poetry'
+      - name: Install mopidy-tidal
+        run: |
+          poetry install
       - name: Lint
         run: |
-          pip install isort black
```
Comment (tehkillerbee) [inline .github/workflows/test.yml:26]: So if I understand correctly, poetry completely takes care of these dependencies now?

File: .gitignore [hunk 2]
```diff
@@ -1,13 +1,176 @@
-*.egg-info
-*.pyc
-*.swp
-.cache/
-.coverage
-.tox/
-MANIFEST
+# Created by https://www.toptal.com/developers/gitignore/api/python
```
Comment (tehkillerbee) [inline .gitignore:1]: Neat! That will come in handy at work!

File: README.md
Comment (tehkillerbee) [inline README.md:1]: We might consider splitting up our README now that the development section has become more detailed.

Perhaps it is time to split it into multiple markdown files. I've wanted to do that with the CHANGELOG as well.

For now, lets merge it into the README, I will create a new issue for a README cleanup.

**Review by tehkillerbee — APPROVED** [review]
Looks great! Only minor comments that can be handled in a future issue.

File: .github/workflows/test.yml (hunk 1 above)
Comment (2e0byo) [reply .github/workflows/test.yml:26]: Yes!  They're in the dev group installed by default but not in the build

File: README.md
Comment (2e0byo) [reply README.md:1]: Or we could have a really small static site?  something like [this](https://2e0byo.github.io/gallica-autobib/) (shameless plug for one of my own projects).  Or even readthedocs, although I'm not a *massive* fan of sphinx in small projects: I'd rather just read the source.

Comment (2e0byo) [reply README.md:1]: oops, moving to issue.

**Pull request merged**